### PR TITLE
Prepare for Pytest v8

### PR DESCRIPTION
**Pull request by ksunden**
## PR summary

The behavior of pytest.warns has changed, particularly with regards to handling of additional warnings

When additional warnings that were not tested for are raised, I simply add a `filterwarnings` (some are artifacts of `with` blocks).

And there are a couple that seem are not actually currently warning at all, but had been occluded by a `pytest.errors`.

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Review by QuLogic — APPROVED** [review]

**Comment by dstansby** [comment]
Is any the CI for this PR running with pytest 8 to check that these fixes do indeed work? Or should I install pytest 8 locally to check?

**Comment by ksunden** [comment]
Looks like I need to figure out a way of fixing one particular test so that it works in both pytest 7 and 8... 

I did indeed just install `pip install -U --pre pytest` to test locally. I think the scheduled prerelease ci run may catch it, but I don't think we have that set up to run on PR via a label like we do for some other CI runs.

**Comment by ksunden** [comment]
Yeah, that particular test raises 2 separate warnings on the same line of code, pytest 7 swallows the outer one and so you get "did not warn", pytest 8 passes it through, so if you _don't_ have both it gets raised... I think I just need to filter the deprecation warning here as it is already tested earlier in the test.

File: lib/matplotlib/backends/backend_pdf.py [hunk 1]
```diff
@@ -2708,6 +2708,7 @@ def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
+        print(f"Closing file {self._filename}")
```
Comment (tacaswell) [inline lib/matplotlib/backends/backend_pdf.py:2711]:
```suggestion
```

debugging print made it through?

**Review by tacaswell — APPROVED** [review]

**Review by QuLogic — APPROVED** [review]

**Pull request merged**